core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.controller.ts

```diff
@@ -21,47 +21,80 @@ require('services/suggestion-modal.service.ts');
 
 angular.module('oppia').controller(
   'TranslationSuggestionReviewModalController', [
-    '$scope', '$uibModalInstance', 'SiteAnalyticsService',
-    'SuggestionModalService', 'contentHtml', 'reviewable', 'translationHtml',
+    '$scope', '$uibModalInstance', 'ContributionAndReviewService',
+    'SiteAnalyticsService', 'SuggestionModalService',
+    'initialSuggestionId', 'reviewable', 'suggestionIdToSuggestion',
     function(
-        $scope, $uibModalInstance, SiteAnalyticsService, SuggestionModalService,
-        contentHtml, reviewable, translationHtml) {
-      $scope.translationHtml = translationHtml;
-      $scope.contentHtml = contentHtml;
+        $scope, $uibModalInstance, ContributionAndReviewService,
+        SiteAnalyticsService, SuggestionModalService,
+        initialSuggestionId, reviewable, suggestionIdToSuggestion) {
+      var resolvedSuggestionIds = [];
       $scope.reviewable = reviewable;
-      $scope.commitMessage = '';
-      $scope.reviewMessage = '';
+      $scope.activeSuggestionId = initialSuggestionId;
+      $scope.activeSuggestion = suggestionIdToSuggestion[
+        $scope.activeSuggestionId];
+      delete suggestionIdToSuggestion[initialSuggestionId];
+      var remaningSuggestions = Object.entries(suggestionIdToSuggestion);
 
       if (reviewable) {
-        // eslint-disable-next-line max-len
-        SiteAnalyticsService.registerContributorDashboardViewSuggestionForReview(
-          'Translation');
+        SiteAnalyticsService
+          .registerContributorDashboardViewSuggestionForReview('Translation');
       }
 
-      $scope.accept = function() {
+      var generateCommitMessage = function() {
+        var contentId = $scope.activeSuggestion.change.content_id;
+        var stateName = $scope.activeSuggestion.change.state_name;
+        var contentType = contentId.split('_')[0];
+
+        return `${contentType} section of "${stateName}" card`;
+      };
+
+      var init = function() {
+        $scope.lastSuggestionToReview = remaningSuggestions.length <= 0;
+        $scope.translationHtml = (
+          $scope.activeSuggestion.change.translation_html);
+        $scope.contentHtml = (
+          $scope.activeSuggestion.change.content_html);
+        $scope.reviewMessage = '';
+      };
+
+      init();
+
+      var showNextItemToReview = function(suggestionId) {
```

Comment:
Does this mean that one can review all available translation suggestions from a single modal? Won't that be confusing since from the list view the user would assume each list item corresponds to one suggestion?

Reply:
Yeah, this was asked by a reviewer and this is better for reviewers.